### ISSUE TITLE
Gradient Descent with Momentum

### ISSUE THREAD

**Issue by Tshimanga** [Deep6AI/epoch#2]
Right now only vanilla gradient descent(backprop) is available as a optimizer. Gradient Descent with Momentum almost always improves the rate of learning. This should be implemented as an alternative option.

**Comment by Tshimanga** [comment]
Done.

commit 663bdd8741bde5bb32b38af2ff7d50a0bc8572e0
